src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/advanced/AddressComponent.tsx

```diff
@@ -88,81 +113,54 @@ export function AddressComponent(props: IAddressComponentProps) {
       }
     };
 
-    fetchPostPlace(props.formData.zipCode, source.token);
-    return function cleanup() {
-      source.cancel('ComponentWillUnmount');
-    };
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [props.formData.zipCode]);
-
-  const onBlurField: (key: AddressKeys, value: any) => void = (key: AddressKeys, value: any) => {
-    const validationErrors: IAddressValidationErrors = validate();
-    setValidations(validationErrors);
-    if (!validationErrors[key]) {
-      props.handleDataChange(value, key);
-      if (key === AddressKeys.zipCode && !value) {
-        // if we are removing a zip code, also remove post place from form data
-        onBlurField(AddressKeys.postPlace, '');
-      }
+    if (
+      prevZipCode.current !== formData.zipCode &&
+      formData.zipCode?.match(/^\d{4}$/)
+    ) {
+      fetchPostPlace();
     }
-  };
 
-  const validate: () => IAddressValidationErrors = () => {
-    const validationErrors: IAddressValidationErrors = {
-      zipCode: null,
-      houseNumber: null,
-      postPlace: null,
+    return () => {
+      source.cancel();
     };
-    if (zipCode && !zipCode.match(new RegExp('^[0-9]{4}$'))) {
-      validationErrors.zipCode = getLanguageFromKey(
-        'address_component.validation_error_zipcode', props.language,
-      );
-      setPostPlace('');
-    } else {
-      validationErrors.zipCode = null;
-    }
-    if (houseNumber && !houseNumber.match(new RegExp('^[a-z,A-Z]{1}[0-9]{4}$'))) {
-      validationErrors.houseNumber = getLanguageFromKey(
-        'address_component.validation_error_house_number', props.language,
+  }, [formData.zipCode, handleDataChange, setZipCodeError]);
```

Comment:
Why will you trigger a new fetch of zipcode info just because any field on the page changes (thus causing handleDataChange to change reference)?

Reply:
Sorry, I see that you handled that case. The issue i see is that you cancel and retry the request if it is not finished quickly and another render of the form is triggered. I think that might cause a horrible slowdown if multiple ajax components are loaded on a single page on a slow connection and only one of the fetches is allowed to finish at one time.

Reply:
Yep, there is an issue here with reredering too quickly and aborting the request. I was sure I had verified this working as it should, but apparently not. Thanks for raising the issue, I will look into it again.

Reply:
I dont think the abort logic is needed here after all, so I removed it

Reply:
Why don't you think that is required for altinn? The implementation in hooks is pretty simple and removing it does not fix neither stale closures or the risk of saving stale results.

Reply:
Not entirely sure what you are asking about here...

Anyway, the previous implementation did have some issues as you pointed out earlier. My other changes caused some weird edge cases I could not figure out, and I dont think its worth spending time on right now - the important thing here was to get rid of the incorrect useEffect dependencies. So I have instead made some minor changes, and kept more of the original code.